pkg/sentry/socket/plugin/cgo/socket_unsafe.go

```diff
@@ -0,0 +1,445 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cgo
+
+/*
+#include <stdint.h>
+#include <sys/epoll.h>
+#include <sys/socket.h>
+
+// epoll
+int plugin_epoll_create(void);
+int plugin_epoll_ctl(int epfd, int op, int fd, struct epoll_event *event);
+int plugin_epoll_wait(int epfd, struct epoll_event *events, int maxevents, int timeout);
+
+// socket control operations
+int plugin_socket(int domain, int type, int protocol, uint64_t *err);
+int plugin_listen(int sockfd, int backlog, uint64_t *err);
+int plugin_bind(int sockfd, const struct sockaddr *addr, socklen_t addrlen, uint64_t *err);
+int plugin_accept(int sockfd, struct sockaddr *addr, socklen_t *addrlen, uint64_t *err);
+int plugin_connect(int sockfd, const struct sockaddr *addr, socklen_t addrlen, uint64_t *err);
+int plugin_getsockopt(int sockfd, int level, int optname,
+			void *optval, socklen_t *optlen, uint64_t *err);
+int plugin_setsockopt(int sockfd, int level, int optname,
+			const void *optval, socklen_t optlen, uint64_t *err);
+int plugin_getsockname(int sockfd, struct sockaddr *addr, socklen_t *addrlen, uint64_t *err);
+int plugin_getpeername(int sockfd, struct sockaddr *addr, socklen_t *addrlen, uint64_t *err);
+int plugin_ioctl(int fd, uint64_t *err, unsigned long int request, void *buf);
+int plugin_shutdown(int sockfd, int how, uint64_t *err);
+int plugin_close(int fd);
+int plugin_readiness(int fd, int events);
+
+// receiving data operations
+ssize_t plugin_recv(int sockfd, void *buf, size_t len, int flags, uint64_t *err);
+ssize_t plugin_recvfrom(int sockfd, void *buf, size_t len, int flags,
+			struct sockaddr *src_addr, socklen_t *addrlen, uint64_t *err);
+ssize_t plugin_recvmsg(int sockfd, struct msghdr *msg, int flags, uint64_t *err);
+ssize_t plugin_read(int fd, void *buf, size_t count, uint64_t *err);
+ssize_t plugin_readv(int fd, const struct iovec *iov, int iovcnt, uint64_t *err);
+
+// sending data operations
+ssize_t plugin_send(int sockfd, const void *buf, size_t len, int flags, uint64_t *err);
+ssize_t plugin_sendto(int sockfd, const void *buf, size_t len, int flags,
+		const struct sockaddr *dest_addr, socklen_t addrlen, uint64_t *err);
+ssize_t plugin_sendmsg(int sockfd, const struct msghdr *msg, int flags, uint64_t *err);
+ssize_t plugin_write(int fd, const void *buf, size_t count, uint64_t *err);
+ssize_t plugin_writev(int fd, const struct iovec *iov, int iovcnt, uint64_t *err);
+*/
+import "C"
+import (
+	"syscall"
+	"unsafe"
+
+	"gvisor.dev/gvisor/pkg/abi/linux"
+	linuxerrno "gvisor.dev/gvisor/pkg/abi/linux/errno"
+)
+
+// EpollCreate works as a CGO wrapper for plugin_epoll_create.
+func EpollCreate() int {
+	return int(C.plugin_epoll_create())
+}
+
+// EpollCtl works as a CGO wrapper for plugin_epoll_ctl.
+func EpollCtl(epfd int32, op int, handle uint32, w uint64) {
```

Comment:
Why not just have w be a uint32? And maybe call it `event` or something for readability?